js/generateFormComponents.js

```diff
@@ -296,6 +296,16 @@ async function createFormComponents() {
 
 	components = createAllComponents(jsonData);
 
+	//Form text box to input GitHub API Key
+	components.push({
+		"label": "GitHub API Key (optional)",
```

Comment:
I would add a description to the component explaining how to get a key, maybe a link to instructions/docs on how to get an API key

Reply:
This should be in the description about token permissions, found from https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#create-a-reference:

> The fine-grained token must have at least one of the following permission sets:
> 
> "Contents" repository permissions (write)
> "Contents" repository permissions (write) and "Workflows" repository permissions (write)

Reply:
One last thing to add, this token permission is needed to create the PR

> The fine-grained token must have the following permission set:
> 
> "Pull requests" repository permissions (write)
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request

Reply:
After doing some testing, here are the necessary repository permissions needed for the token:
- Contents: read &write
- Workflows: read & write
- Pull requests: read & write

Reply:
I added this as a description:


>   Generate a Github API Key from here: https://github.com/settings/tokens/new .
>		
>    The token should have these permissions: 
>			- Contents: read &write 
>			- Workflows: read & write
>			- Pull requests: read & write